src/client/common/application/encryptedStorage.ts

```diff
@@ -34,8 +34,14 @@ export class EncryptedStorage implements IEncryptedStorage {
         if (IS_REMOTE_NATIVE_TEST && this.extensionContext.extensionMode !== ExtensionMode.Production) {
             return this.testingState.get(`${service}#${key}`);
         }
-        // eslint-disable-next-line
-        const val = await this.extensionContext.secrets.get(`${service}.${key}`);
-        return val;
+        try {
+            // eslint-disable-next-line
+            const val = await this.extensionContext.secrets.get(`${service}.${key}`);
+            return val;
+        } catch (e) {
+            // If we get an error trying to get a secret, it might be corrupted. So we delete it.
+            await this.extensionContext.secrets.delete(`${service}.${key}`);
```

Comment:
According to this issue https://github.com/microsoft/vscode/issues/120558, an encryption change in Windows requires us to re enter every secret.

I checked and this is the only place in the repo where .get() gets called. So if we get an error while getting, we can delete and then enter the secret again as usual.

Thanks @IanMatthewHuff for the help.

Reply:
We considered what other secret could be affected by this, but it looks like its only the remote uri's.